unit/testing-utils/invariant.cpp

```diff
@@ -28,3 +28,11 @@ std::string invariant_failure_containingt::describe() const
          expected + "\"";
 }
 
+std::ostream &
+operator<<(std::ostream &out, const invariant_failedt &invariant_failed)
```

Comment:
Would this be useful generally with `invariant_failedt` rather than in testing utils?

Reply:
I am not sure. This printing seems fairly specific to the unit tests. If I put it with `invariant_failedt` then I think it would increase the risk that someone might unwittingly rename it to `pretty_print` or something rather than leaving it as an operator; which would revert the unit test printing back to using `{?}`, without giving any kind of error.